4.homework(multithreading_basics)/1.task#1/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func main() {
+	// you can choose the file name if you need
+	fileName := "output.csv"
+
+	dataChannel := make(chan string)
+	signalChannel := make(chan os.Signal, 1)
+	stopChannel := make(chan struct{})
+
+	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
+
+	createFile(fileName)
+
+	go reader(dataChannel)
+	go writer(dataChannel, fileName, stopChannel)
+
+	<-signalChannel
+	fmt.Println("Program completed successfully.")
+	close(stopChannel)
+	close(dataChannel)
+}
+
+func reader(ch chan<- string) {
+	scanner := bufio.NewScanner(os.Stdin)
+	for {
+		fmt.Println("Write a line that is added to the file: ")
+		if scanner.Scan() {
+			input := scanner.Text()
+			ch <- input
+		} else {
+			break
+		}
+	}
+}
+
+func writer(ch <-chan string, fileName string, stop <-chan struct{}) {
+	file, err := os.OpenFile(fileName, os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		fmt.Println("File not found: ", err)
+		return
+	}
+	defer closeFile(file)
+
+	for {
+		select {
+		case input, ok := <-ch:
+			if !ok {
+				return
+			}
+			_, err := file.WriteString(input + "\n")
+			if err != nil {
+				fmt.Println("File writing error: ", err)
+				return
+			}
+		case <-stop:
+			return
+		}
+	}
+}
+
+func createFile(fileName string) {
+	if _, err := os.Stat(fileName); os.IsNotExist(err) {
+		file, err := os.Create(fileName)
+		if err != nil {
+			fmt.Println("File creating error: ", err)
+			return
+		}
+		defer closeFile(file)
+	} else if err != nil {
+		fmt.Println("Error with file: ", err)
+		return
+	}
```

Comment:
чисто по стилистике: когда обрабатываешь несколько разных вариантов чуть красивее и читабельнее выглядит свитч
```suggestion
	_, err := os.Stat(fileName)
	switch {
	case os.IsNotExist(err):
		file, err := os.Create(fileName)
		if err != nil {
			fmt.Println("File creating error: ", err)
			return
		}
		defer closeFile(file)
	case err != nil:
		fmt.Println("Error with file: ", err)
	}
```